### PR TITLE
Use questionary buttons when interacting with assistant via the terminal

### LINKED ISSUE #3020: Allow users to select buttons when running from cmdline
When an utterance has buttons, we list them like:
```
1: ...
2: ...
```
Would be nice to allow users to select these by, for instance, typing "1" or "2".  Or using questionary

### PR DESCRIPTION
**Pull request by paulaWesselmann**
**Proposed changes**:
- if the bot message contains buttons, the user will be able to chose a reply from a list choice instead of having to type the buttons payload 
- Fixes #3020 
- Works in both interactive learning mode and with `rasa shell`

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by MetcalfeTom** [comment]
@tabergma let me know if you have a better solution for printing the buttons during restore mode:
https://github.com/RasaHQ/rasa/pull/3487/files#diff-65df5d45a8db1437674b14a313bbf414R100

Otherwise the tests fail with the following:
https://travis-ci.com/RasaHQ/rasa/jobs/202226954#L4054

**Comment by tabergma** [comment]
@MetcalfeTom Do we have any example bot with buttons that I can use to try this out?

**Comment by MetcalfeTom** [comment]
`moodbot`

**Comment by tabergma** [comment]
Ok, for me the solution does not seem to work yet. Will have a close look after lunch.

Screenshot 2019-05-23 at 10 06 18

File: rasa/core/channels/console.py [hunk 1]
```diff
@@ -50,14 +58,20 @@ def print_bot_output(message, color=rasa.cli.utils.bcolors.OKBLUE):
...
 
 
-def get_cmd_input():
-    response = questionary.text(
-        "", qmark="Your input ->", style=Style([("qmark", "#b373d6"), ("", "#b373d6")])
-    ).ask()
+def get_cmd_input(button_question: questionary.Question):
+    if button_question:
+        text = button_question.ask()
+        response = text[text.find("(") + 1 : text.find(")")]  # get intent
```
Comment (tabergma) [inline rasa/core/channels/console.py]: Can we extract that into a separate method as the exact code is also used in interactive learning?

File: rasa/core/channels/console.py [hunk 2]
```diff
@@ -50,14 +58,20 @@ def print_bot_output(message, color=rasa.cli.utils.bcolors.OKBLUE):
         )
 
 
-def get_cmd_input():
-    response = questionary.text(
-        "", qmark="Your input ->", style=Style([("qmark", "#b373d6"), ("", "#b373d6")])
-    ).ask()
+def get_cmd_input(button_question: questionary.Question):
+    if button_question:
```
Comment (tabergma) [inline rasa/core/channels/console.py]: Can we make it more expressive? I would initialize `button_question` with `None` and also return `None` in `print_bot_output` at the end. The check here I would modify to `button_question is not None`.

File: rasa/core/channels/console.py [hunk 3]
```diff
@@ -50,14 +58,20 @@ def print_bot_output(message, color=rasa.cli.utils.bcolors.OKBLUE):
         )
 
 
-def get_cmd_input():
-    response = questionary.text(
-        "", qmark="Your input ->", style=Style([("qmark", "#b373d6"), ("", "#b373d6")])
-    ).ask()
+def get_cmd_input(button_question: questionary.Question):
```
Comment (tabergma) [inline rasa/core/channels/console.py]: Add return type

File: rasa/core/channels/console.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 
 def print_bot_output(message, color=rasa.cli.utils.bcolors.OKBLUE):
```
Comment (tabergma) [inline rasa/core/channels/console.py]: Add return type

File: rasa/core/restore.py [hunk 5]
```diff
@@ -94,8 +94,17 @@ def actions_since_last_utterance(tracker: DialogueStateTracker) -> List[Text]:
                 event.text, sender_id=tracker.sender_id, output_channel=out
             )
             for m in out.messages:
+                buttons = m.pop("buttons", None)  # for non-terminal stdin
                 console.print_bot_output(m)
 
+                if buttons:
```
Comment (tabergma) [inline rasa/core/restore.py]: I prefer `if buttons is not None`.

**Comment by MetcalfeTom** [comment]
Thank you @paulaWesselmann for your contribution and for pushing to get this issue fixed.

**Pull request merged**